File: app/src/main/java/com/mithril/mobilegoldenleaf/models/Category.java
package com.mithril.mobilegoldenleaf.models;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity
public class Category {
    @PrimaryKey(autoGenerate = true)
    private int id;
    private  String title;


    @Ignore
    public Category(String title) {
        this.title = title;
    }

    @Ignore
    public Category(int id, String title) {
        this.id = id;
        this.title = title;
    }

    public Category() {

    }


    @Override
    public String toString() {
        return "Category{" +
                "title='" + title + '\'' +
                '}';
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }
}

File: app/src/main/java/com/mithril/mobilegoldenleaf/models/Category.kt
package com.mithril.mobilegoldenleaf.models

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity
class Category {
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0
    var title: String? = null

    constructor()

    @Ignore
    constructor(title: String) {
        this.title = title
    }

    @Ignore
    constructor(id: Long, title: String) {
        this.id = id
        this.title = title
    }


    override fun toString(): String {
        return "Category{" +
                "title='" + title + '\''.toString() +
                '}'.toString()
    }
}
